### PR TITLE
feat(locale): add Korean language support and translations

### PR DESCRIPTION
**Pull request by phungmanhcuong**
### Summary
This pull request introduces full Korean language support to enhance accessibility for Korean-speaking users.

### Changes Made
- **Translations**:
  - Added `ko.json` containing comprehensive Korean translations for the application.
- **Locale Configuration**:
  - Updated `AVAILABLE_LOCALES` in `sharedConstants.ts` to include `ko`.
  - Extended the `Locale` type in `users.ts` to recognize the Korean locale.

### Impact
- Users can now interact with the application in Korean, improving accessibility and user experience for Korean-speaking audiences.

### Testing
- Verified proper loading of translations for the Korean locale.
- Confirmed updates to locale configurations do not impact existing functionality.
- Ensured no regressions in locale-based features.

### Notes
- It's important to thoroughly verify the translations in all Korean message files to ensure accuracy and clarity. 
- Further testing in staging is recommended to validate the new locale in a production-like environment.

### CODE REVIEW

File: client/locales/ko.json [hunk 1]
```diff
@@ -0,0 +1,7892 @@
...
+  "course.assessment.submission.runCodeWithLimit": {
+    "defaultMessage": "코드 실행 ({attemptsLeft, plural, one {# 시도} other {# 시도}} 남음)"
+  },
+  "course.assessment.submission.saveDraft": {
+    "defaultMessage": "초안 저장"
+  },
+  "course.assessment.submission.saveGrade": {
+    "defaultMessage": "등급 저장"
+  },
+  "course.assessment.submission.savedAt": {
```
Comment (cysjonathan) [inline client/locales/ko.json:3281]: #9659 removed some translations.
Ensure these are removed in yours too.
https://github.com/Coursemology/coursemology2/pull/7659/files#diff-2e6669fb7d6cf2ec9e23dd3fda6a1f8bf7efd4be5764c04a99c47d72f5f0c118

**Review by cysjonathan — CHANGES_REQUESTED** [review]

File: config/locales/ko/course/assessment/answer/programming_auto_grading.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+              timeout_error: >
+                작업 실행 시간이 길어져 타임아웃 오류가 발생했습니다. 몇 분 후에 다시 시도해 주세요. 문제가
+                계속되면 강사나 coursemology 관리자에게 문의하세요.
+              container_unreachable: >
+                평가기를 위한 도커 생성 중 오류가 발생했습니다. 서버 작업 부하가 높기 때문일
+                수 있습니다. 몇 분 후에 다시 시도해 주세요. 문제가 계속되면 강사나 coursemology 관리자에게 문의하세요.
+              generic_error: >
+                오류가 발생했습니다. 서버가 바쁠 수 있으니 몇 분 후에 다시 시도해 보세요. 동일한 문제가
+                계속되면, coursemology 관리자에게 문의하고 이 페이지의 URL을 제공하며 다음 오류를 함께 제공해주세요: '"%{error}"
+
```
Comment (cysjonathan) [inline config/locales/ko/course/assessment/answer/programming_auto_grading.yml:31]: remove trailing spaces? Just leave a single trailing newline.

File: config/locales/ko/notifiers/course/assessment/answer/comment_notifier/annotated/user_notifications/email.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+            annotated:
+              user_notifications:
+                email:
+                  subject: '%{course}: %{topic}에 대한 새 주석'
+                  message: >
+                    <p>%{post_author}님이 주석을 추가했습니다: %{post}</p>
+                    
+                    <p>여기에서 답장할 수 있습니다: </p> 
+                    %{topic}
+
```
Comment (cysjonathan) [inline config/locales/ko/notifiers/course/assessment/answer/comment_notifier/annotated/user_notifications/email.yml:16]: single trailing newline not two. Check for all your files.

**Review by cysjonathan — CHANGES_REQUESTED** [review]

File: config/locales/ko/course/assessment/answer/programming_auto_grading.yml (hunk 2 above)
Comment (phungmanhcuong) [reply config/locales/ko/course/assessment/answer/programming_auto_grading.yml:31]: Thats the syntax of "folded block scalar." of yaml: https://yaml-multiline.info/

**Review by cysjonathan — APPROVED** [review]

**Pull request merged**